src/components/duration/time.rs

```diff
@@ -198,19 +200,20 @@ impl TimeDuration {
         // 12. Return ! CreateTimeDurationRecord(days × sign, hours × sign, minutes × sign, seconds × sign, milliseconds × sign, microseconds × sign, nanoseconds × sign).
         let days = (days as i64).mul_add(sign.into(), 0);
         let result = Self::new_unchecked(
-            (hours as i32).mul_add(sign, 0).into(),
-            (minutes as i32).mul_add(sign, 0).into(),
-            (seconds as i32).mul_add(sign, 0).into(),
-            (milliseconds as i32).mul_add(sign, 0).into(),
-            (microseconds as i32).mul_add(sign, 0).into(),
-            (nanoseconds as i32).mul_add(sign, 0).into(),
+            FiniteF64::from((hours as i32).mul_add(sign, 0)),
```

Comment:
This sounds like a perfect fit for [copysign](https://doc.rust-lang.org/stable/std/primitive.f64.html#method.copysign).